compiler/packages/babel-plugin-react-compiler/src/Optimization/OutlineJsx.ts

```diff
@@ -230,21 +232,13 @@ function collectProps(
         return null;
       }
 
-      /*
-       * TODO(gsn): Handle attributes that have same value across
-       * the outlined jsx instructions.
-       */
-      if (seen.has(at.name)) {
-        return null;
-      }
-
       if (at.kind === 'JsxAttribute') {
-        seen.add(at.name);
         attributes.push({
           originalName: at.name,
-          newName: at.name,
+          newName: seen.has(at.name) ? `${at.name}${id++}` : at.name,
```

Comment:
nit: This will break if there are multiple `a` props and an `a1` prop. In practice it's unlikely, but let's update to guarantee that we generate a fresh name in all cases.